# NBA All-Star Weekend Challenge

Your task...answer the following questions given the dataset given about the NBA All-Star Game from the year 2000 - 2016

Become more familiar with the dataset <a href="https://www.kaggle.com/fmejia21/nba-all-star-game-20002016">here</a>

Once completed, commit this notebook to github and submit the link to the google classroom assignment.

<a href="https://classroom.google.com/u/2/c/NDc4MzEzMjI5Nzla/a/NTE3OTYxNzM2OTNa/details">Google Classroom</a>

=========================================================================================================================

In [1]:
import pandas as pd

In [2]:
# storing the csv data into a pandas data frame
nba_data = pd.read_csv('NBA All Stars 2000-2016 - Sheet1.csv')
nba_data

Year           Player Pos    HT   WT                    Team  \
0    2016    Stephen Curry   G   6-3  190   Golden State Warriors   
1    2016     James Harden  SG   6-5  220         Houston Rockets   
2    2016     Kevin Durant  SF   6-9  240   Golden State Warriors   
3    2016    Kawhi Leonard   F   6-7  230       San Antonio Spurs   
4    2016    Anthony Davis  PF  6-11  253    New Orleans Pelicans   
..    ...              ...  ..   ...  ...                     ...   
434  2000  Antonio McDyess   F   6-9  245          Denver Nuggets   
435  2000      Gary Payton   G   6-4  180     Seattle SuperSonics   
436  2000  Rasheed Wallace  FC  6-11  230  Portland Trail Blazers   
437  2000   David Robinson   C   7-1  250       San Antonio Spurs   
438  2000      Vlade Divac   C   7-1  260        Sacramento Kings   

                             Selection Type    NBA Draft Status    Nationality  
0       Western All-Star Fan Vote Selection   2009 Rnd 1 Pick 7  United States  
1       Western All-Star Fan Vote Selection   2009 Rnd 1 Pick 3  United States  
2       Western All-Star Fan Vote Selection   2007 Rnd 1 Pick 2  United States  
3       Western All-Star Fan Vote Selection  2011 Rnd 1 Pick 15  United States  
4       Western All-Star Fan Vote Selection   2012 Rnd 1 Pick 1  United States  
..                                      ...                 ...            ...  
434      Western All-Star Coaches Selection   1995 Rnd 1 Pick 2  United States  
435      Western All-Star Coaches Selection   1990 Rnd 1 Pick 2  United States  
436      Western All-Star Coaches Selection   1995 Rnd 1 Pick 4  United States  
437      Western All-Star Coaches Selection   1987 Rnd 1 Pick 1  United States  
438  Western All-Star Replacement Selection  1989 Rnd 1 Pick 26         Serbia  

[439 rows x 9 columns]

In [3]:
# info about nba_data
nba_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 439 entries, 0 to 438
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Year              439 non-null    int64 
 1   Player            439 non-null    object
 2   Pos               439 non-null    object
 3   HT                439 non-null    object
 4   WT                439 non-null    int64 
 5   Team              439 non-null    object
 6   Selection Type    439 non-null    object
 7   NBA Draft Status  439 non-null    object
 8   Nationality       439 non-null    object
dtypes: int64(2), object(7)
memory usage: 31.0+ KB


In [4]:
nba_data.describe(include = "all")

Year       Player  Pos    HT          WT        Team  \
count    439.000000          439  439   439  439.000000         439   
unique          NaN          123   11    18         NaN          34   
top             NaN  Kobe Bryant    G  6-11         NaN  Miami Heat   
freq            NaN           16   72    72         NaN          28   
mean    2007.974943          NaN  NaN   NaN  228.751708         NaN   
std        4.899148          NaN  NaN   NaN   30.834435         NaN   
min     2000.000000          NaN  NaN   NaN  165.000000         NaN   
25%     2004.000000          NaN  NaN   NaN  210.000000         NaN   
50%     2008.000000          NaN  NaN   NaN  235.000000         NaN   
75%     2012.000000          NaN  NaN   NaN  246.000000         NaN   
max     2016.000000          NaN  NaN   NaN  325.000000         NaN   

                            Selection Type    NBA Draft Status    Nationality  
count                                  439                 439            439  
unique                                   6                 123             22  
top     Western All-Star Coaches Selection  1996 Rnd 1 Pick 13  United States  
freq                                   119                  16            365  
mean                                   NaN                 NaN            NaN  
std                                    NaN                 NaN            NaN  
min                                    NaN                 NaN            NaN  
25%                                    NaN                 NaN            NaN  
50%                                    NaN                 NaN            NaN  
75%                                    NaN                 NaN            NaN  
max                                    NaN                 NaN            NaN

In [5]:
nba_unique = {}
for row in nba_data.keys():
    nba_unique[row] = len(nba_data[row].unique())

nba_unique

{'Year': 17,
 'Player': 123,
 'Pos': 11,
 'HT': 18,
 'WT': 49,
 'Team': 34,
 'Selection Type': 6,
 'NBA Draft Status': 123,
 'Nationality': 22}

 - nba_data - stores data about players in the NBA All-Star Game from the year 2000 - 2016.
 - there are 439 rows, with no "null" values
 - there are 9 columns with the following # of unique values and data type.
     - Year: 17, integer
     - Player: 123, object (string)
     - Pos: 11, object (string)
     - HT: 18, object (number separated by -)
     - WT: 49, integer
     - Team: 34, object (string)
     - Selection Type: 6, object (string)
     - NBA Draft Status: 123, object (string, Number Rnd Number Pick Number)
     - Nationality: 22, object (string)

# What is the average weight of all players who played during this time?

### Average weight of ALL players

In [53]:
all_avg_weight = nba_data["WT"].mean()
print(f"Average of ALL weights of ALL players during the entire 2000 - 2016 time period is {round(all_avg_weight)} lbs.")

Average of ALL weights of ALL players during the entire 2000 - 2016 time period is 229 lbs.


### Average weight of players each year

In [56]:
avg_weight_per_year = nba_data.groupby('Year', as_index = False).mean().rename(columns = {"WT": "Average Weight"})
avg_weight_per_year

Year  Average Weight
0   2000      228.928571
1   2001      229.153846
2   2002      231.200000
3   2003      233.375000
4   2004      231.833333
5   2005      228.640000
6   2006      229.655172
7   2007      228.576923
8   2008      228.961538
9   2009      225.964286
10  2010      231.400000
11  2011      227.320000
12  2012      227.600000
13  2013      227.840000
14  2014      226.214286
15  2015      227.346154
16  2016      225.458333

### Average weight of each player who played during this time

In [48]:
pd.set_option('display.max_rows', None)

avg_weight = nba_data.groupby('Player', as_index = False).mean().rename(columns = {"WT": "Average Weight"})
avg_weight[['Player','Average Weight']]

Player  Average Weight
0               Al Horford           245.0
1            Allan Houston           205.0
2            Allen Iverson           165.0
3          Alonzo Mourning           240.0
4        Amar'e Stoudemire           245.0
5           Andre Drummond           279.0
6           Andre Iguodala           215.0
7         Andrei Kirilenko           235.0
8             Andrew Bynum           285.0
9           Antawn Jamison           235.0
10           Anthony Davis           253.0
11           Anthony Mason           250.0
12          Antoine Walker           265.0
13           Antonio Davis           245.0
14         Antonio McDyess           245.0
15             Baron Davis           215.0
16             Ben Wallace           240.0
17           Blake Griffin           251.0
18             Brad Miller           261.0
19             Brandon Roy           211.0
20             Brook Lopez           268.0
21           Carlos Boozer           266.0
22         Carmelo Anthony           240.0
23            Caron Butler           228.0
24        Chauncey Billups           210.0
25              Chris Bosh           235.0
26             Chris Kaman           265.0
27              Chris Paul           175.0
28            Chris Webber           245.0
29          Damian Lillard           195.0
30           Danny Granger           222.0
31               David Lee           245.0
32          David Robinson           250.0
33              David West           250.0
34          DeAndre Jordan           265.0
35           DeMar DeRozan           220.0
36        DeMarcus Cousins           270.0
37          Deron Williams           200.0
38            Derrick Rose           190.0
39            Devin Harris           185.0
40         Dikembe Mutombo           260.0
41           Dirk Nowitzki           245.0
42          Draymond Green           230.0
43           Dwight Howard           265.0
44             Dwyane Wade           220.0
45             Elton Brand           254.0
46             Gary Payton           180.0
47          Gerald Wallace           220.0
48   Giannis Antetokounmpo           222.0
49          Gilbert Arenas           191.0
50          Glenn Robinson           225.0
51          Gordon Hayward           226.0
52              Grant Hill           225.0
53           Isaiah Thomas           185.0
54         Jamaal Magloire           259.0
55          Jamal Mashburn           247.0
56           Jameer Nelson           190.0
57            James Harden           220.0
58              Jason Kidd           210.0
59             Jeff Teague           186.0
60         Jermaine O'Neal           235.0
61        Jerry Stackhouse           218.0
62            Jimmy Butler           231.0
63             Joakim Noah           230.0
64             Joe Johnson           240.0
65               John Wall           210.0
66             Josh Howard           210.0
67            Jrue Holiday           205.0
68             Karl Malone           265.0
69           Kawhi Leonard           230.0
70            Kemba Walker           184.0
71           Kenyon Martin           240.0
72            Kevin Durant           240.0
73           Kevin Garnett           240.0
74              Kevin Love           251.0
75           Klay Thompson           215.0
76             Kobe Bryant           212.0
77             Kyle Korver           212.0
78              Kyle Lowry           205.0
79            Kyrie Irving           193.0
80       LaMarcus Aldridge           240.0
81        Latrell Sprewell           190.0
82            LeBron James           250.0
83               Luol Deng           220.0
84           Manu Ginobili           205.0
85              Marc Gasol           255.0
86             Mehmet Okur           249.0
87       Metta World Peace           260.0
88          Michael Finley           225.0
89          Michael Jordan           215.0
90            Michael Redd           220.0
91             Mo Williams           198.0
92               Pau 

### double checking the calculation above

In [32]:
stephen_curry = nba_data[nba_data['Player'] == 'Stephen Curry']
print(stephen_curry[["Year","Player", "WT"]])

stephen_curry_avg_weight = avg_weight[avg_weight['Player'] == 'Stephen Curry']
print(stephen_curry_avg_weight)

    Year         Player   WT
0   2016  Stephen Curry  190
41  2015  Stephen Curry  190
63  2014  Stephen Curry  190
87  2013  Stephen Curry  190
            Player    Year     WT
108  Stephen Curry  2014.5  190.0


# What team is represented the most during this time?

### Miami Heat is represented the most, followed by Boston Celtics, San Antonio Spurs and Los Angeles Lakers.

In [58]:
team_count = nba_data.groupby(["Team"], as_index = False).count().rename(columns = {"Player": "Number of Players"})
team_count[["Team", "Number of Players"]].sort_values(by = "Number of Players", ascending = False).head(5)

Team  Number of Players
16          Miami Heat                 28
1       Boston Celtics                 26
29   San Antonio Spurs                 25
14  Los Angeles Lakers                 25
11     Houston Rockets                 20

# What draft class is represented the most during this time?

### 1996 Rnd 1 Pick 13 is represented the most!

In [60]:
draft_class = nba_data.groupby(["NBA Draft Status"], as_index = False).count().rename(columns = {"Player": "Number of Players"})
draft_class[["NBA Draft Status", "Number of Players"]].sort_values(by = "Number of Players", ascending = False).head(5)

NBA Draft Status  Number of Players
26  1996 Rnd 1 Pick 13                 16
68   2003 Rnd 1 Pick 1                 13
35   1997 Rnd 1 Pick 1                 13
42   1998 Rnd 1 Pick 9                 13
23   1995 Rnd 1 Pick 5                 12

# Are foreign players more prevalent in the All-Star Game during this time?

In [67]:
nationalities = nba_data["Nationality"].unique()
print(nationalities)

['United States' 'Spain' 'United States\nAustralia' 'Greece\nNigeria'
 'Dominican Republic\nUnited States' 'Germany' 'France'
 'France\nUnited States' 'England\nSouth Sudan' 'Canada\nSouth Africa'
 'China' 'Argentina' 'United States\nGermany' 'Germany\nUnited States'
 'Turkey' 'Lithuania' 'Russia' 'Serbia\nGreece' 'Canada'
 'Democratic Republic of the Congo' 'United States\nSpain' 'Serbia']


### For this analysis, I defined foreign players as those without United States citizenship/nationality.

### There are more players with US Citizenship in the All-Star Game during this time.

In [76]:
nba_data ["Citizenship"] = ["US" if "United States" in x else "Foreign" for x in nba_data["Nationality"]]
us_citizenship = nba_data.groupby("Citizenship", as_index = False).count().rename(columns = {"Player":"Number of Players"})
us_citizenship[["Citizenship", "Number of Players"]]

Citizenship  Number of Players
0     Foreign                 60
1          US                379

# How often are the Western All-Stars voted in by fans?

### 39% of the time

In [107]:
selection = nba_data.groupby("Selection Type", as_index = False).count().rename(columns = {"Player":"Number of Players"})
selection = selection[["Selection Type", "Number of Players"]]
print(selection)

                           Selection Type  Number of Players
0      Eastern All-Star Coaches Selection                119
1     Eastern All-Star Fan Vote Selection                 85
2  Eastern All-Star Replacement Selection                 16
3      Western All-Star Coaches Selection                119
4     Western All-Star Fan Vote Selection                 85
5  Western All-Star Replacement Selection                 15


In [108]:
western_all_star = selection["Number of Players"][3:].sum()
western_fans = selection["Number of Players"][4]

print(western_fans)
print(western_all_star)

print(f'Western All-Stars were voted in by fans {round(100 * western_fans/western_all_star)}% of the time.')


85
219
Western All-Stars were voted in by fans 39% of the time.


# How many times does Steph Curry make the All-Star game during this time period?

### 4 times!

In [110]:
curry = nba_data.loc[nba_data["Player"] == "Stephen Curry"]
curry

Year         Player Pos   HT   WT                   Team  \
0   2016  Stephen Curry   G  6-3  190  Golden State Warriors   
41  2015  Stephen Curry   G  6-3  190  Golden State Warriors   
63  2014  Stephen Curry   G  6-3  190  Golden State Warriors   
87  2013  Stephen Curry   G  6-3  190  Golden State Warriors   

                         Selection Type   NBA Draft Status    Nationality  \
0   Western All-Star Fan Vote Selection  2009 Rnd 1 Pick 7  United States   
41  Western All-Star Fan Vote Selection  2009 Rnd 1 Pick 7  United States   
63  Western All-Star Fan Vote Selection  2009 Rnd 1 Pick 7  United States   
87  Western All-Star Fan Vote Selection  2009 Rnd 1 Pick 7  United States   

   US Citizenship Citizenship  
0             Yes          US  
41            Yes          US  
63            Yes          US  
87            Yes          US

# How many Shooting Guards have made the All-Star game during this time?

### 57 Shooting Guards made the All-Star game!

In [122]:
print(nba_data["Pos"].unique())
sgs = nba_data.groupby("Pos", as_index = False).count().rename(columns = {"Player":"Number of Players"})
sgs = sgs.loc[sgs["Pos"] == "SG"]
sgs[["Pos", "Number of Players"]]

['G' 'SG' 'SF' 'F' 'PF' 'C' 'GF' 'PG' 'FC' 'F-C' 'G-F']


Pos  Number of Players
10  SG                 57

# How many Chicago Bulls players have made the All-Star Game during this time period?

### 12 Chicago Bulls Players made the All-Star Game!

In [121]:
by_team = nba_data.groupby("Team", as_index = False).count().rename(columns = {"Player":"Number of Players"})
bulls = by_team.loc[by_team["Team"] == "Chicago Bulls"]
bulls[["Team", "Number of Players"]]

Team  Number of Players
5  Chicago Bulls                 12